alica_engine/include/engine/blackboard/BlackBoard.h

```diff
@@ -15,49 +11,25 @@ namespace alica
 class BlackBoard
 {
 public:
-    using IdType = BBIdent;
-    using ObjectType = ByteArray;
     BlackBoard() = default;
 
-    IdType registerValue(const int8_t* buffer, int len);
-    IdType registerValue(const char* buffer, int len);
-
-    template <class InputIt>
-    IdType registerValue(InputIt begin, InputIt end);
-
-    bool hasValue(IdType id) const { return _body.find(id) != _body.end(); }
+    void registerValue(const std::string& key, std::any any);
+    void registerValue(const char* key, std::any any);
```

Comment:
can't think of any way to do this because the type information is lost after the any object is constructed & subsequently it will have to be supplied as a template parameter to know it. However we can't use templates since setTag etc. is a virtual method

Reply:
doesn't std::any object give you type information?

Reply:
It gives runtime type info, which can't be used in templates

Reply:
we can have a check on the type and do explicit typecasting about supported type. Or at least do std::string conversion inside the AlicaTracing object rather then making every caller do a string conversion at the place of call.